include/nbl/builtin/hlsl/bxdf/ndf/microfacet_to_light_transform.hlsl

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BXDF_NDF_MICROFACET_LIGHT_TRANSFORM_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BXDF_NDF_MICROFACET_LIGHT_TRANSFORM_INCLUDED_
+
+#include "nbl/builtin/hlsl/limits.hlsl"
+#include "nbl/builtin/hlsl/bxdf/common.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace bxdf
+{
+namespace ndf
+{
+
+enum MicrofacetTransformTypes : uint16_t
+{
+    MTT_REFLECT = 0b01,
+    MTT_REFRACT = 0b10,
+    MTT_REFLECT_REFRACT = 0b11
+};
+
+template<typename T, bool IsGgx, MicrofacetTransformTypes reflect_refract>
+struct SDualMeasureQuant;
+
+template<typename T>
+struct SDualMeasureQuant<T, false, MTT_REFLECT>
+{
+    using this_t = SDualMeasureQuant<T, false, MTT_REFLECT>;
+    using scalar_type = T;
+
+    NBL_CONSTEXPR_STATIC_INLINE MicrofacetTransformTypes Type = MTT_REFLECT;
+
+    scalar_type getMicrofacetMeasure()
+    {
+        return pdf;
+    }
+
+    // this computes the max(NdotL,0)/(4*max(NdotV,0)*max(NdotL,0)) factor which transforms PDFs in the f in projected microfacet f * NdotH measure to projected light measure f * NdotL
+    scalar_type getProjectedLightMeasure()
+    {
+        return scalar_type(0.25) * pdf / maxNdotV;
+    }
+
+    scalar_type pdf;
```

Comment:
I would not call this `pdf` cause sometimes its an importance sampled value, so just `value`